Sources/Security.php

```diff
@@ -955,18 +959,24 @@ function allowedTo($permission, $boards = null, $any = false)
 				break;
 		}
 		$smcFunc['db_free_result']($request);
+		$return = $result;
 		return $result;
 	}
 
 	// Make sure they can do it on all of the boards.
-	if ($smcFunc['db_num_rows']($request) != count($boards))
-		return false;
+	if ($smcFunc['db_num_rows']($request) != count($boards) && empty($return))
+		$return = false;
 
-	$result = true;
-	while ($row = $smcFunc['db_fetch_assoc']($request))
-		$result &= !empty($row['add_deny']);
-	$smcFunc['db_free_result']($request);
+	if (empty($return))
+	{
```

Comment:
Unless I read this wrong, the previous line may set this as false when it used to return false, but then we start processing this with $result = true and returning $result.  Potentially it could mean we get a false positive here.

Might be better to set $return to null, then check !is_null or $return !== null

Reply:
Yes empty is here wrong, 
I need to use here isset.